inc/zoo/swar/SWAR.h

```diff
@@ -83,99 +88,153 @@ namespace detail {
         }
     };
     template<> struct Selector_impl<true> {
         template<int Level>
-        constexpr static uint64_t execute(uint64_t v) {
+        constexpr static uint64_t execute(uint64_t v) noexcept {
             return popcount_builtin<Level - 2>(v);
         }
     };
 
 }
 
 template<int Level>
-constexpr uint64_t popcount(uint64_t a) {
+constexpr uint64_t popcount(uint64_t a) noexcept {
     return detail::Selector_impl<2 < Level>::template execute<Level>(a);
 }
 
-static_assert(0x210 == popcount<0>(0x320), "");
-static_assert(0x4321 == popcount<1>(0xF754), "");
-static_assert(0x50004 == popcount<3>(0x3E001122), "");
-
 
-template<typename T> constexpr typename std::make_unsigned<T>::type msb_index(T v) {
+template<typename T> constexpr typename std::make_unsigned<T>::type msb_index(T v) noexcept {
     return 8*sizeof(T) - 1 - __builtin_clzll(v);
 }
 
-template<int Size, typename T = uint64_t> struct SWAR {
+template<typename T> constexpr typename std::make_unsigned<T>::type lsb_index(T v) noexcept {
+    return __builtin_ctzll(v) + 1;
+}
+
+template<int NBits, typename T = uint64_t> struct SWAR {
     SWAR() = default;
     constexpr explicit SWAR(T v): m_v(v) {}
+    constexpr explicit operator T() const noexcept { return m_v; }
 
     constexpr T value() const noexcept { return m_v; }
 
-    constexpr SWAR operator|(SWAR o) { return SWAR(m_v | o.m_v); }
-    constexpr SWAR operator&(SWAR o) { return SWAR(m_v & o.m_v); }
-    constexpr SWAR operator^(SWAR o) { return SWAR(m_v ^ o.m_v); }
+    constexpr SWAR operator|(SWAR o) const noexcept { return SWAR(m_v | o.m_v); }
+    constexpr SWAR operator&(SWAR o) const noexcept { return SWAR(m_v & o.m_v); }
+    constexpr SWAR operator^(SWAR o) const noexcept { return SWAR(m_v ^ o.m_v); }
 
-    constexpr bool operator==(SWAR o) { return m_v == o.m_v; }
-
-    constexpr T at(int position) {
-        constexpr auto filter = (T(1) << Size) - 1;
-        return filter & (m_v >> (Size * position));
+    constexpr T at(int position) const noexcept {
+        constexpr auto filter = (T(1) << NBits) - 1;
+        return filter & (m_v >> (NBits * position));
     }
 
-    constexpr SWAR clear(int position) {
-        constexpr auto filter = (T(1) << Size) - 1;
-        auto invertedMask = filter << (Size * position);
+    constexpr SWAR clear(int position) const noexcept {
+        constexpr auto filter = (T(1) << NBits) - 1;
+        auto invertedMask = filter << (NBits * position);
         auto mask = ~invertedMask;
         return SWAR(m_v & mask);
     }
 
-    constexpr int top() { return msb_index(m_v) / Size; }
-    constexpr int fastIndex() { return __builtin_ctzll(m_v) / Size; }
+    /// Notably, this is the SWAR lane index that contains the MSB.
+    constexpr int top() const noexcept { return msb_index(m_v) / NBits; }
+    /// Notably, this is the SWAR lane index that contains the lsb.
+    constexpr int lsbIndex() const noexcept { return __builtin_ctzll(m_v) / NBits; }
```

Comment:
What is the meaning of "Notably" here?

Reply:
I might adopt your implicit suggestion of calling SWAR elements "lanes"

Reply:
as opposed to msb_index/lsb_index that find those indexes against a u64 or other integral type.

Reply:
Still not agreeing with "notably"

Reply:
Some other writer around SWAR/SIMD calls them lanes, but I can't remember where I first saw it.  Seems like a useful mental model.